src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1624,18 +1624,20 @@ public async Task Step(int threadId, enum_STEPKIND kind, enum_STEPUNIT unit)
 
             await ExceptionManager.EnsureSettingsUpdated();
 
+            bool isForwardStep = kind != enum_STEPKIND.STEP_BACKWARDS;
             if ((unit == enum_STEPUNIT.STEP_LINE) || (unit == enum_STEPUNIT.STEP_STATEMENT))
             {
                 switch (kind)
                 {
                     case enum_STEPKIND.STEP_INTO:
-                        await MICommandFactory.ExecStep(threadId);
+                        await MICommandFactory.ExecStep(threadId, isForwardStep);
```

Comment:
instead of using `isForwardStep` check the `exeuctionDirection`

Reply:
`STEP_BACKWARDS`is a valid input though.

Reply:
It is an older enum from `IDebugProcess3` and that is only for stepping. We now use the newer interface `IDebugReversibleSessionProcess160` to handle reverse program control which includes `ReverseStep` and `ReverseExecute`.